dot/network/service.go

```diff
@@ -742,3 +743,43 @@ func (s *Service) startProcessingMsg() {
 		}
 	}
 }
+
+func (s *Service) BlockAnnounceHandshake(header *types.Header) error {
+	peers := s.host.peers()
+	if len(peers) < 1 {
```

Comment:
can you write it as `len(peers) == 0`?

Reply:
I think the preference for using `len(peers) < 1` over `len(peers) == 0` is because error tolerance. This is safety if there is an error somewhere in the code.

There is a concept: [Off-by-one error](https://en.wikipedia.org/wiki/Off-by-one_error) and using `<` could prevent it

Reply:
On second thought, since the list can't have a negative len, it might make sense to compare with 0 😅